### PR TITLE
Canonicalize `tlangs` to lowercase

### LINKED ISSUE #1925: `Locale` not canonicalizing `tlangs` to lowercase
`Locale` canonicalizes `tlangs` (the "-t-en-latn" part inside the "en-t-en-latn" locale) incorrectly, preserving the canonicalized case instead of converting all to lowercase.

The [RFC6497](https://www.ietf.org/rfc/rfc6497.txt) from the IETF that describes the extension T (and tlangs) specifies on one of its paragraphs:

```
2.3.  Canonicalization

   As required by [BCP47], the use of uppercase or lowercase letters is
   not significant in the subtags used in this extension.  The canonical
   form for all subtags in the extension is lowercase, with the fields
   ordered by the separators, alphabetically.  The order of subtags
   within a field is significant, and MUST NOT be changed in the process
   of canonicalizing.
```

Emphasis on 

> The canonical form for all subtags in the extension is lowercase ...

### Minimized snippet with the bug

```Rust
let test: Locale = "en-t-en-latn-ca-emodeng".parse().unwrap();
println!("{test}"); // prints "en-t-en-Latn-CA-emodeng" instead of "en-t-en-latn-ca-emodeng"
```

Haven't checked if it would be necessary to fix only `Locale::from_bytes` or also `Locale::canonicalize` and `LocaleCanonicalizer::canonicalize`.

### PR DESCRIPTION
**Pull request by jedel1043**
Closes #1925

For simplicity I didn't add `write_to_lowercase` methods for `Script` and `Region`, but let me know if that would be preferable.

### CODE REVIEW

File: components/locid/tests/fixtures/canonicalize.json [hunk 1]
```diff
@@ -14,5 +14,25 @@
...
+    },
+    "output": {
+      "type": "Locale",
+      "identifier": "en-US-t-es-ar-x-foo"
+    }
+  },
+  {
+    "input": {
+      "type": "Locale",
+      "identifier": "en-t-en-Latn-CA-emodeng"
```
Comment (robertbastian) [inline components/locid/tests/fixtures/canonicalize.json:31]: please test uppercase variants

**Review by sffc — APPROVED** [review]
Praise: Clean and simple solution

**Review by sffc — APPROVED** [review]

**Comment by jedel1043** [comment]
Huh, `error: failed to build archive: No space left on device (os error 28)`. Is this a sporadic fail?

**Review by robertbastian — DISMISSED** [review]

**Review by zbraniecki — COMMENTED** [review]
looks good and less invasive than I was worried it would be. Please, add the documentation to help readers understand why we need it (with examples).

**Comment by jedel1043** [comment]
The error seems to be similar to https://github.com/myoung34/docker-github-actions-runner/issues/322, so maybe the container just needs to be removed to free its storage?

**Review by sffc — APPROVED** [review]

**Review by robertbastian — APPROVED** [review]

**Review by zbraniecki — APPROVED** [review]
lgtm!

**Comment by jedel1043** [comment]
Friendly ping to note that this can be merged now.

**Pull request merged**